src/libmongoc/tests/test-mongoc-cursor.c

```diff
@@ -305,8 +309,9 @@ _test_common_opts (void *ctx)
    BSON_ASSERT (mongoc_cursor_set_hint (cursor, sd->id));
    ASSERT_CMPINT (mongoc_cursor_get_hint (cursor), ==, sd->id);
 
-   /* listDatabases prohibits limit and batchSize */
-   if ((make_cursor_fn) ctx != _make_array_cursor) {
+   /* listDatabases and isMaster prohibits limit and batchSize */
```

Comment:
Added "isMaster". 

The issue was caused by the if guards changed below. `_make_cmd_deprecated_cursor` creates a cursor with an "isMaster" command. Since "isMaster" commands - like the "listDatabases" referenced here - do not accept a "batchSize" or "limit", the server errored with the message (paraphrased) "batchSize (or limit) option not found".